vcd/datasource_vcd_nsxv_dhcp_relay.go

```diff
@@ -0,0 +1,74 @@
+package vcd
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceVcdNsxvDhcpRelay() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceVcdNsxvDhcpRelayRead,
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name for DHCP relay settings",
+			},
+			"ip_addresses": {
+				Computed:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP address of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"domain_names": {
+				Computed:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP domain names of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ip_sets": {
+				Computed:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP set names which consist DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"relay_agent": {
+				Computed: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"org_network": {
```

Comment:
Two comments:

1.) It seems, in H5 UI in addition to Org networks you can select external network:

![Screen Shot 2019-12-06 at 16 30 20](https://user-images.githubusercontent.com/11330689/70330183-b82d2d80-1845-11ea-8cb9-2afbc48c0ac6.png)

Are you sure it's only org networks?

2.) I'm also thinking whether we shouldn't rename `org_network` -> `org_network_name` for clarity. It would be similar to `network_name` in `vcd_nsxv_snat` resource.

3.) And continuing on the train of thought of (1) and (2), if it turns out that we can select external network, then we could just name the field `network_name` as in NAT rules.

Reply:
> Two comments:
> 
> 1.) It seems, in H5 UI in addition to Org networks you can select external network:
> 
> ![Screen Shot 2019-12-06 at 16 30 20](https://user-images.githubusercontent.com/11330689/70330183-b82d2d80-1845-11ea-8cb9-2afbc48c0ac6.png)
> 
> Are you sure it's only org networks?
Yes. It does allow to choose, but here is what happens:
<img width="960" alt="image" src="https://user-images.githubusercontent.com/15804230/70332262-d85eeb80-1849-11ea-926c-ef8068e036c1.png">

> 
> 2.) I'm also thinking whether we shouldn't rename `org_network` -> `org_network_name` for clarity. It would be similar to `network_name` in `vcd_nsxv_snat` resource.
> 
Can do. I kind of thought between this and that. In some places we don't ask for "name", in some we do. (like `edge_gateway` also asks for name, but we don't mention it).

> 3.) And continuing on the train of thought of (1) and (2), if it turns out that we can select external network, then we could just name the field `network_name` as in NAT rules.

I can make it broader - maybe in future it works although for my limited understanding this wouldn't make sense. The general point of relay is to "forward" DHCP request messages to defined servers outside of NSX. One shouldn't be able to forward DHCP request messages from external network. I might be wrong though.

Reply:
OK, so we have three choices:
- `org_network`
- `org_network_name`
- `network_name`

@Didainius @dataclouder @vbauzysvmware your votes?

To me `network_name` looks most consistent with the same field from NSX-V NAT rules. However, it doesn't reflect the point that these are only Org networks.

Reply:
`network_name` is more generic and will work even if we end up supporting different kinds of network

Reply:
I am fine with `network_name`